### DISCUSSION TITLE
how to use waypoint plugin's %% waypoint %% in template?

### DISCUSSION THREAD

**Discussion by francisHugo** [SilentVoid13/Templater#1329] [Help] [answered]
Hi, 

I am using template and waypoint plugin for a while, and both of them are terrific for my use.

now I would like to put waypoint trigger into a template note, meaning, to produce a folder note by using a template.   

however,  when I put the `%% waypoint %%` (to trigger dynamic folder file list in a folder note) into a template, it feedback an err says: *%% Error: Cannot create a waypoint in a note that's not the folder note. For more information, check the instructions [here](https://github.com/IdreesInc/Waypoint) %%*

**Question**

Is possible to put `%% waypoint %%` in a template note? (while waypoint plugin is enabled)

BR

**Comment by Zachatoo** [comment]
Is it showing an error in your template, or in a note after applying the template?

The error seems to be saying that the waypoint code only works if the note name matches its parent folder name. Are you fulfilling that requirement?

**Comment by francisHugo** [comment]
The error message is showing in my template, and also in a note after applying the template.

I noticed that the waypoint code only works with the note name matches its parent's folder name. However, I would like to have some specific text I wrote together with the waypoint code (`%% waypoint %%`) in a template note, so I can generate a folder notes with the template and the contents which the waypoint code generated.

I tried several ways but no luck to make it works.

**Reply by Zachatoo** [reply]
Are you trying to generate a folder and a folder note? Or do you have an existing folder that you're trying to add a folder note for? What's the workflow?

**Reply by francisHugo** [reply]
No, not to generate a folder and a folder note, only for folder note.  My folders are already existing, for each folder there is already a note keeps why the folder exist, and the name of that note is not identical to the folder name. 

I would like to have a template for folder notes, the template would looks like this:

`starts of the template`

```
---
tags: foldernotes
---

![[embed why the folder exist]]

file list:
%% waypoint %%
```
`end of the template`


There is another plugin named `folder notes` allows to use template to generate folder notes. I use it to generate the folder notes while planning to apply the above template. 

The reason for using  `folder notes` is that with it, can get control to when the  folder note should be opened(I set to use Alt + mouse click). I used to use a plugin `folder note` (without S, the author of the plugin is xpgo, it does not upede for 3 years, I am little afraid it will not compatible with future obsidian, so I am trying to swich to `folder notes`). in `folder note`, I can put the `%% waypoint %%` in its `setting options` like 

![image](https://github.com/SilentVoid13/Templater/assets/129242259/74a455c9-1be6-4cfb-9366-2ceb58e36641)


it works well. The problem is that when the folder contains large amount of files, it will take times to generate the file list, and I could not control whether just to collapse the folder or open the folder note.

**Reply by Zachatoo** [reply, marked as answer]
Reading the [docs](https://lostpaul.github.io/obsidian-folder-notes/Getting%20started/) for the folder notes plugin, it looks like you can set a template for any newly created folder notes. Then you can use a folder notes command or context menu option to create a new folder note for a folder with a given template. The folder notes plugin should make sure that the newly created file matches the name of the folder, and then the waypoint code should work.

**Reply by francisHugo** [reply]
Thank you for your times and helps. I manually input it after applying the template now. 

anyway, thank you very much for your help.

BR

**Reply by offbrands-studio** [reply]
I found a different solution to activate [Waypoint](https://github.com/IdreesInc/Waypoint) and using the newer plugin [Folder Notes](https://github.com/LostPaul/obsidian-folder-notes) plugin @francisHugo 

```
# [[<% tp.file.title %>]]

%% Waypoint %<% tp.file.cursor(1) %>%
```

If set up this way, it will activate whenever you create a Folder Note without the initial template  activating. (I have my default Folder Template listed with a Heading and a link to the file title, this isn't necessary, just my preference.)

You can use Templater to create a Hotkey for Waypoint as well if you want more control. 

`%% Waypoint %<% tp.file.cursor(1) %>%`

- Create a Template with only this in the body of the note 
- Set up the Hotkey in Templater settings 
- Pick keys to create (I used CTRL CMD W)
- After creating a Folder Note via Folder Notes Plugin, insert the command via the Hotkey or Command Menu, and you should be good to go. 

I hope I've explained this well enough, if not, I'll try to expand if anyone has questions.

**Reply by AusardiaTx** [reply]
You can directly input `%% Begin Waypoint %%` and `%% End Waypoint %%`(Can be omitted) instead of `%% Waypoint %%`in a template.

**Comment by chenbihao** [comment]
it works for me:
<%*
const waypoint = "%% Way" + "point %%";
tR += waypoint;
%>


from https://github.com/IdreesInc/Waypoint/issues/90#issuecomment-2175639979
